cmd/snap-confine/mount-support-nvidia.c

```diff
@@ -433,10 +447,26 @@ static int sc_mount_nvidia_is_driver_in_dir(const char *dir)
 	if (driver.major_version == 0) {
 		return 0;
 	}
+	// Extra sanity check
+	if (driver.glcore_pattern == NULL) {
+		die("libnvidia-glcore pattern is unset");
+	}
+
 	// Probe if a well known library is found in directory dir
-	sc_must_snprintf(driver_path, sizeof driver_path,
-			 "%s/" LIBNVIDIA_GLCORE_SO_PATTERN, dir,
-			 driver.major_version, driver.minor_version);
+	// cant concat `%s/` and `driver.glcore_pattern` since it's not a constant
+	if (sc_streq(driver.glcore_pattern, LIBNVIDIA_GLCORE_SO_PATTERN)) {
```

Comment:
i know it's not pretty
given that `snprintf("%s/" driver.glcore_pattern)` doesn't work, can you suggest something better @bboozzoo?